frontend_tests/zjsunit/zjquery.js

```diff
@@ -114,8 +114,17 @@ exports.make_zjquery = function () {
                 return self.wrapper;
             },
             on: function (name, f) {
-                var funcs = on_functions.setdefault(name, []);
-                funcs.push(f);
+                function add_func_to_on_event(name, f) {
+                    var funcs = on_functions.setdefault(name, []);
```

Comment:
Aditya and I discussed this on chat in some depth, so just updating the PR for any other person reading this.  We are gonna support multiple calling variations of `on`, including passing in selectors, and then we'll add a new method to support code like this:

~~~ js
var event_name = 'click';
var child_selector = '#my-child';
var f = $('#my-parent').get_on_handler(event_name, child_selector);
f(....);